async_producer_test.go

```diff
@@ -1695,12 +1715,16 @@ func TestAsyncProducerInterceptors(t *testing.T) {
 		},
 		{
 			name:         "interceptor chain",
-			interceptors: []ProducerInterceptor{&appendInterceptor{i: 0}, &appendInterceptor{i: 1000}},
+			interceptors: []ProducerInterceptor{&appendInterceptor{i: 0}, &appendInterceptor{i: 1000}, &metadataInterceptor{text: "sarama"}},
 			expectationFn: func(t *testing.T, i int, msg *ProducerMessage) {
 				v, _ := msg.Value.Encode()
-				expected := TestMessage + strconv.Itoa(i) + strconv.Itoa(i+1000)
-				if string(v) != expected {
-					t.Errorf("Interceptor should have incremented the value, got %s, expected %s", v, expected)
+				expectedValue := TestMessage + strconv.Itoa(i) + strconv.Itoa(i+1000)
```

Comment:
There’s not really a reason we need to disambiguate `expected` here. There’s only one expected value here, so `expected` is a reasonable name for it.

Reply:
this actually controls the value field, maybe it would be better if I change the `str` variable below it to `expectedMetadata`. What do you think?

Reply:
I updated the code according to the message above, please let me know if you want to go with `expected`. I will update to it again.